framework/src/postprocessors/GeneralSensorPostprocessor.C

```diff
@@ -0,0 +1,315 @@
+  uncertainty_value = MooseRandom::randNormal(0, uncertainty_std_dev);
+  
+  //-------------DELAY-------------
+  // delay (tau) as a function of time
+  Real delay_value = 0.0;
+  delay_value = _delay_function.value(_t);
+  
+  // delayed input signal
+  Real t_desired = _t - delay_value;
+  Real _input_signal_delayed;
+  Real _input_signal_initial;
+  Real for_int_initial;
+
+  if (_t == 0)
+  {
+    _input_signal_initial = _input_signal;
+    for_int_initial = _input_signal; //_input_signal_delayed + signalToNoise_value * normal_value;
+  }
+
+  if (t_desired < 0)
+  {
+    _input_signal_delayed = 0;
+  }
+
+  else if (t_desired == 0)
+  {
+    _input_signal_delayed = _input_signal_initial;
+  }
+
+  // linear interpolation
+  else if (t_desired > 0 && t_desired <= _t && t_desired >= _t - _dt)
+  {
+    _input_signal_delayed = _input_signal + (t_desired - _t) * (_pp_old - _input_signal) / (-_dt);
+  }
+
+  else if (t_desired > 0 && t_desired < _t - _dt && floor(t_desired/_dt)*_dt == 0)
+  {
+    _input_signal_delayed = _input_signal_initial + t_desired * (_input_signal_values[0] - _input_signal_initial) / _time_values[0];
+  }
+
+  else if (t_desired > 0 && t_desired < _t - _dt && floor(t_desired/_dt)*_dt != 0)
+  {
+    _input_signal_delayed = linearInterpolationInVectors(_time_values, _input_signal_values, t_desired);
+  }
+
+  else
+  { 
+    cout << "Some error in interpolation values." <<endl;
+  }
+
+
+  //-------------INTEGRAL-------------
+  Real for_int = (_input_signal_delayed + signalToNoise_value * 1) ; // + signalToNoise_value * normal_value;
+  _for_int.push_back(for_int);
+  
+  // New vector for integration for this timestep only
+  std::vector<Real> __for_int = _for_int;
+  std::vector<Real> __time_values = _time_values;
+  std::vector<Real> _for_exp;
+  Real integration_value;
+  std::vector<Real> integrand;
+  
+  for (size_t i = 0; i < _time_values.size(); ++i) 
+    {
+        _for_exp.push_back(exp(-(_t - _time_values[i])/delay_value)); 
+    }
+  
+    // exact exponential at delay time
+    Real _for_exp_at_delay = exp(-(_t - delay_value)/delay_value);
+
+  if (_t > delay_value && delay_value > _dt)
+  {
+    // linearly interpolation of for_int at delay time
+    Real _for_int_at_delay = linearInterpolationInVectors(__time_values, __for_int, delay_value);
+    // push the delay and interpolated signal value into the new vector
+    __for_int.push_back(_for_int_at_delay);
+    __time_values.push_back(delay_value);
+    _for_exp.push_back(_for_exp_at_delay);
+    // sort the vector in aascending order
+    std::sort(__for_int.begin(), __for_int.end());
+    std::sort(__time_values.begin(), __time_values.end());
+    std::sort(_for_exp.begin(), _for_exp.end());
+    // New vector for integration with all elements before delay removed from __for_int and _for_exp
+    removeValuesBeforeTime(__time_values, __for_int, _for_exp, delay_value); 
+
+    integrand = elementwiseMultiply(__for_int, _for_exp);

```

Comment:
```suggestion
    integrand = __for_int.dot(_for_exp);
```

and the types of  `__for_int` and `_for_exp` should be `DenseVector<Real>` (also please remove the leading `_` for local vars)

Reply:
I cannot find any DenseVector files, there is a DenseMatrix

Reply:
`#include "libmesh/dense_vector.h"`

Reply:
addressed.